tools/block-generator/generator/generate.go

```diff
@@ -398,10 +405,20 @@ func (g *generator) WriteBlock(output io.Writer, round uint64) error {
 		if err != nil {
 			return err
 		}
-		g.finishRound(numTxnForBlock)
+
+		// only advance the round if the requestor is caught up
```

Comment:
We have this logic to account for genesis block being requested during Init()

Reply:
TODO: timebox possibilty of caching a round.

Reply:
```suggestion
		// only advance the round if the requestor is caught up.
		// we have this logic to account for an extra block request made by conduit during startup.
```

Reply:
I ended up refactoring the code to cache the previous round's msgp. The intent of this suggested comment lives elsewhere. I'll point it out below.